diff --git a/common/se_wrapper/Makefile b/common/se_wrapper/Makefile
index d741a7b9..cc550c82 100644
--- a/common/se_wrapper/Makefile
+++ b/common/se_wrapper/Makefile
@@ -31,7 +31,7 @@
 
 include ../../buildenv.mk
 
-CFLAGS += -Werror -D_GNU_SOURCE -fPIC
+CFLAGS += -D_GNU_SOURCE -fPIC
 CFLAGS += $(ADDED_INC)
 
 CPPFLAGS := -I$(COMMON_DIR)/inc \
diff --git a/common/se_wrapper_psw/Makefile b/common/se_wrapper_psw/Makefile
index e776c2b2..2f7cc4a3 100644
--- a/common/se_wrapper_psw/Makefile
+++ b/common/se_wrapper_psw/Makefile
@@ -31,7 +31,7 @@
 
 include ../../buildenv.mk
 
-CFLAGS += -Werror -D_GNU_SOURCE -fPIC
+CFLAGS += -D_GNU_SOURCE -fPIC
 CFLAGS += $(ADDED_INC)
 
 CPPFLAGS := -I$(SGX_HEADER_DIR) \
diff --git a/psw/ae/le/Makefile b/psw/ae/le/Makefile
index 525f154d..579000e9 100644
--- a/psw/ae/le/Makefile
+++ b/psw/ae/le/Makefile
@@ -34,9 +34,6 @@ include ../buildenv.mk
 
 AENAME = launch_enclave
 
-CFLAGS   += -Werror
-CXXFLAGS += -Werror
-
 INCLUDE += -I$(LINUX_PSW_DIR)/ae/common
 INCLUDE += -I$(LINUX_PSW_DIR)/ae/data/constants/linux 
 INCLUDE += -I$(SGX_HEADER_DIR)/libcxx
diff --git a/psw/ae/pce/Makefile b/psw/ae/pce/Makefile
index d9d349d2..8d841456 100644
--- a/psw/ae/pce/Makefile
+++ b/psw/ae/pce/Makefile
@@ -33,9 +33,6 @@ TOP_DIR := ../../..
 MITIGATION-CVE-2020-0551=LOAD
 include ../buildenv.mk
 
-CXXFLAGS  += -Werror
-CFLAGS    += -Werror 
-
 AENAME = pce
 
 INCLUDE += -I$(LINUX_PSW_DIR)/ae/data/constants/linux 
diff --git a/psw/ae/pve/Makefile b/psw/ae/pve/Makefile
index 8fb76caf..2e134087 100644
--- a/psw/ae/pve/Makefile
+++ b/psw/ae/pve/Makefile
@@ -33,8 +33,6 @@ TOP_DIR := ../../..
 MITIGATION-CVE-2020-0551=LOAD
 include ../buildenv.mk
 
-CXXFLAGS  += -Werror
-CFLAGS	  += -Werror 
 AENAME = provision_enclave
 
 INCLUDE += -I$(LINUX_PSW_DIR)/ae/data/constants/linux 
@@ -57,7 +55,7 @@ OBJS := $(sort $(OBJ) version.o)
 
 AENAME_OUT := $(AENAME).so
 
-EPID_FILTER = -Werror -nostdinc -nostdinc++
+EPID_FILTER = -nostdinc -nostdinc++
 EPID_CFLAGS = $(filter-out $(EPID_FILTER),$(ENCLAVE_CFLAGS))
 EPID_CXXFLAGS = $(filter-out $(EPID_FILTER),$(ENCLAVE_CXXFLAGS))
 
diff --git a/psw/ae/qe/Makefile b/psw/ae/qe/Makefile
index 9571f932..a712bcd4 100644
--- a/psw/ae/qe/Makefile
+++ b/psw/ae/qe/Makefile
@@ -33,9 +33,6 @@ TOP_DIR := ../../..
 MITIGATION-CVE-2020-0551=LOAD
 include ../buildenv.mk
 
-CXXFLAGS  += -Werror
-CFLAGS	+= -Werror
-
 EXTERNAL_LIB += -L$(EPID_SDK_DIR)/epid/member -lmember \
 				-L$(EPID_SDK_DIR)/epid/common -lcommon
 
@@ -53,7 +50,7 @@ SRC += se_sig_rl.cpp                  \
 OBJ := $(SRC:.cpp=.o)
 OBJS := $(sort $(OBJ) version.o)
 
-EPID_FILTER = -Werror -nostdinc -nostdinc++
+EPID_FILTER = -nostdinc -nostdinc++
 EPID_CFLAGS = $(filter-out $(EPID_FILTER),$(ENCLAVE_CFLAGS))
 EPID_CXXFLAGS = $(filter-out $(EPID_FILTER),$(ENCLAVE_CXXFLAGS))
 
diff --git a/psw/ae/ref_le/Makefile b/psw/ae/ref_le/Makefile
index f2b653dd..5688d0a9 100644
--- a/psw/ae/ref_le/Makefile
+++ b/psw/ae/ref_le/Makefile
@@ -43,9 +43,6 @@ REF_WL_GEN_DIR   = ./ref_wl_gen
 
 WL_VERSION       = 1
 
-CFLAGS   += -Werror
-CXXFLAGS += -Werror
-
 INCLUDE += -I$(LINUX_PSW_DIR)/ae/common
 INCLUDE += -I$(LINUX_PSW_DIR)/ae/data/constants/linux
 INCLUDE += -I$(LINUX_PSW_DIR)/ae/inc/internal
diff --git a/psw/ae/ref_le/ref_wl_gen/Makefile b/psw/ae/ref_le/ref_wl_gen/Makefile
index 4c4180a8..2ae0882a 100644
--- a/psw/ae/ref_le/ref_wl_gen/Makefile
+++ b/psw/ae/ref_le/ref_wl_gen/Makefile
@@ -42,9 +42,6 @@ CPPFLAGS    += -I$(SGX_HEADER_DIR)                      \
             -I$(LINUX_SDK_DIR)/sign_tool/SignTool/      \
             -I$(LINUX_EXTERNAL_DIR)/rdrand
 
-CXXFLAGS += -Werror
-CFLAGS += -Werror
-
 RDRAND_LIBDIR := $(LINUX_EXTERNAL_DIR)/rdrand/src
 RDRAND_MAKEFILE := $(RDRAND_LIBDIR)/Makefile
 
diff --git a/psw/enclave_common/Makefile b/psw/enclave_common/Makefile
index 6627e99c..8167f8c4 100644
--- a/psw/enclave_common/Makefile
+++ b/psw/enclave_common/Makefile
@@ -38,10 +38,10 @@ CXXFLAGS += -DDISABLE_TRACE
 CFLAGS += -DDISABLE_TRACE
 endif
 
-CXXFLAGS += -fPIC -Werror -g
+CXXFLAGS += -fPIC -g
 CXXFLAGS += $(ADDED_INC)
 
-CFLAGS += -fPIC -Werror -g
+CFLAGS += -fPIC -g
 CFLAGS += $(ADDED_INC)
 
 INC += -I$(SGX_HEADER_DIR)                \
diff --git a/psw/uae_service/linux/Makefile b/psw/uae_service/linux/Makefile
index c797b00a..956c6973 100644
--- a/psw/uae_service/linux/Makefile
+++ b/psw/uae_service/linux/Makefile
@@ -62,7 +62,7 @@ INCLUDE += -I$(LINUX_EXTERNAL_DIR)/epid-sdk \
            -I$(LINUX_PSW_DIR)/ae/aesm_service/source \
            -I$(LINUX_PSW_DIR)/ae/aesm_service/source/common
 
-CXXFLAGS += -fPIC -Werror -Wno-unused-parameter -g -DPROTOBUF_INLINE_NOT_IN_HEADERS=0
+CXXFLAGS += -fPIC -Wno-unused-parameter -g -DPROTOBUF_INLINE_NOT_IN_HEADERS=0
 
 EXTERNAL_LIB += -lprotobuf 
 
@@ -181,7 +181,7 @@ messages.pb.o : $(IPC_COMMON_PROTO_DIR)/messages.pb.cc
 %.o :%.cpp
 	$(CXX) $(CXXFLAGS) -Wno-deprecated-declarations $(INCLUDE) -c $< -o $@
 %.o: %.c
-	$(CC) $(CFLAGS) $(INCLUDE) -Werror -fPIC -c $< -o $@
+	$(CC) $(CFLAGS) $(INCLUDE) -fPIC -c $< -o $@
 
 $(BUILD_DIR):
 	@$(MKDIR) $@
diff --git a/psw/urts/linux/Makefile b/psw/urts/linux/Makefile
index 529d7c2b..f9261199 100644
--- a/psw/urts/linux/Makefile
+++ b/psw/urts/linux/Makefile
@@ -36,10 +36,10 @@ CXXFLAGS += -DDISABLE_TRACE
 CFLAGS += -DDISABLE_TRACE
 endif
 
-CXXFLAGS += -fPIC -Werror -g
+CXXFLAGS += -fPIC -g
 CXXFLAGS += $(ADDED_INC)
 
-CFLAGS += -fPIC -Werror -g
+CFLAGS += -fPIC -g
 CFLAGS += $(ADDED_INC)
 
 VTUNE_DIR = $(LINUX_EXTERNAL_DIR)/vtune/linux
diff --git a/psw/urts/parser/Makefile b/psw/urts/parser/Makefile
index 9edcdecc..610f0973 100644
--- a/psw/urts/parser/Makefile
+++ b/psw/urts/parser/Makefile
@@ -49,10 +49,6 @@ TARGET := libenclaveparser.a
 
 CXXFLAGS += -fPIC
 
-ifneq ($(HOST_OS), darwin)
-  CXXFLAGS += -Werror
-endif
-
 CXXFLAGS += -I$(COMMON_DIR)/inc/         \
             -I$(COMMON_DIR)/inc/internal \
             -I.                          \
