diff --git a/psw/enclave_common/Makefile b/psw/enclave_common/Makefile
index 6627e99c..73e16ee8 100644
--- a/psw/enclave_common/Makefile
+++ b/psw/enclave_common/Makefile
@@ -33,10 +33,8 @@ include ../../buildenv.mk
 
 ECL_VER:= $(shell awk '$$2 ~ /ENCLAVE_COMMON_VERSION/ { print substr($$3, 2, length($$3) - 2); }' $(COMMON_DIR)/inc/internal/se_version.h)
 
-ifndef DEBUG
 CXXFLAGS += -DDISABLE_TRACE
 CFLAGS += -DDISABLE_TRACE
-endif
 
 CXXFLAGS += -fPIC -Werror -g
 CXXFLAGS += $(ADDED_INC)
diff --git a/psw/urts/linux/Makefile b/psw/urts/linux/Makefile
index 529d7c2b..0e93db84 100644
--- a/psw/urts/linux/Makefile
+++ b/psw/urts/linux/Makefile
@@ -31,10 +31,8 @@
 
 include ../../../buildenv.mk
 
-ifndef DEBUG
 CXXFLAGS += -DDISABLE_TRACE
 CFLAGS += -DDISABLE_TRACE
-endif
 
 CXXFLAGS += -fPIC -Werror -g
 CXXFLAGS += $(ADDED_INC)
diff --git a/sdk/libcapable/linux/Makefile b/sdk/libcapable/linux/Makefile
index d1f4cba7..de4f7781 100644
--- a/sdk/libcapable/linux/Makefile
+++ b/sdk/libcapable/linux/Makefile
@@ -31,10 +31,8 @@
 
 include ../../../buildenv.mk
 
-ifndef DEBUG
 CXXFLAGS += -DDISABLE_TRACE
 CFLAGS += -DDISABLE_TRACE
-endif
 
 CXXFLAGS += -fPIC -Werror
 
diff --git a/sdk/simulation/uinst/Makefile b/sdk/simulation/uinst/Makefile
index ac6842bd..aa2c091c 100644
--- a/sdk/simulation/uinst/Makefile
+++ b/sdk/simulation/uinst/Makefile
@@ -33,10 +33,8 @@ include ../../../buildenv.mk
 
 SIM_DIR := $(CUR_DIR)/..
 
-ifndef DEBUG
 CXXFLAGS += -DDISABLE_TRACE
 CFLAGS += -DDISABLE_TRACE
-endif
 
 CPPFLAGS += $(ADDED_INC)
 CPPFLAGS += -I$(COMMON_DIR)/inc/         \
diff --git a/sdk/simulation/uinst/linux/Makefile b/sdk/simulation/uinst/linux/Makefile
index 60d5d3b3..c6609b7a 100644
--- a/sdk/simulation/uinst/linux/Makefile
+++ b/sdk/simulation/uinst/linux/Makefile
@@ -33,10 +33,8 @@ include ../../../../buildenv.mk
 
 SIM_DIR := $(CUR_DIR)/../..
 
-ifndef DEBUG
 CXXFLAGS += -DDISABLE_TRACE
 CFLAGS += -DDISABLE_TRACE
-endif
 
 CFLAGS += -Werror -fPIC
 
diff --git a/sdk/simulation/urtssim/linux/Makefile b/sdk/simulation/urtssim/linux/Makefile
index 0d6dcb71..a05f4048 100644
--- a/sdk/simulation/urtssim/linux/Makefile
+++ b/sdk/simulation/urtssim/linux/Makefile
@@ -34,10 +34,8 @@ include ../../../../buildenv.mk
 SIM_DIR := $(LINUX_SDK_DIR)/simulation
 VTUNE_DIR = $(LINUX_EXTERNAL_DIR)/vtune/linux
 
-ifndef DEBUG
 CXXFLAGS += -DDISABLE_TRACE
 CFLAGS += -DDISABLE_TRACE
-endif
 
 CXXFLAGS += -fPIC -DSE_SIM -Werror -g
 CFLAGS   += -fPIC -DSE_SIM -Werror -g
